File: solutions/cpp/grains/1/grains.cpp
#include "grains.h"
#include <cmath>

namespace grains {
    double square(double number) {
        double result = pow(2.0, number - 1);
        return result;
    }
    double total() {
        return pow(2, 64);
    }
}  // namespace grains

File: solutions/python/grains/1/grains.py
def square(number):
    if number < 1 or number > 64:
        raise ValueError("square must be between 1 and 64")
    else:
        grains = 1 * (2 ** (number - 1))
        return grains

def total():
    total = 0

    for square_no in range(1,65):
        total = total + square(square_no)

    return total
